### DISCUSSION TITLE
how would i change the type chart

### DISCUSSION THREAD

**Discussion by SnolidIce** [wavebeem/pkmn.help#215] [Q&A] [answered]
I would like to make my own version of the type chart and this website but i am very new to coding, where would i look at to change the type caculator

**Comment by wavebeem** [comment, marked as answer]
Currently the `data-types` and `data-matchups` files should be the main things that need editing, as far as I can remember.

- https://github.com/wavebeem/pkmn.help/blob/master/src/data-types.ts
  - `Type`
  - `types`
- https://github.com/wavebeem/pkmn.help/blob/master/src/data-matchups.ts
  - `typesInPokemondbOrder`
  - `gen1`, `gen2` etc

Let me know if you run into any trouble editing those files.

**Comment by SnolidIce** [comment]
oh i see ty

**Comment by SnolidIce** [comment]
alright now that i have made my edits how would i test it?

**Reply by wavebeem** [reply]
do the development steps work for you?

https://github.com/wavebeem/pkmn.help#readme

```bash
$ npm install
$ npm start
```

**Reply by wavebeem** [reply]
when you run `npm start` you should see a message about a local URL you can use to test it. it should be like `http://localhost:3000/` or something like that. the number might be different. if you visit that URL in your browser the application should start

**Reply by atendev** [reply]
Hello, I have changed the values in the table for the matchups and re-run npm install and npm start, however i don't ever see a change in the chart. Am I doing something wrong?

**Reply by wavebeem** [reply]
test by changing some text on the page to see if the change is reflected

hard to say without knowing more

**Reply by atendev** [reply]
I was able to figure it out. It was working fine I was just looking at the actual website I had pulled up on a second monitor haha. I apologize. 

But this is sick! I’m working on a fan game with a custom “more balanced” type chart and being able to adjust it and see how each Pokémon in my dex is effected by it is insanely useful. Especially being able to have this available for those who would play the game.

Amazing work dude!

**Comment by SnolidIce** [comment]
yeah i got it running, ty

**Comment by wavebeem** [comment]
Glad to hear it! Send me a link if you feel comfortable sharing it

*Sage Fennel* (they/them)
wavebeem.com** <https://www.wavebeem.com>
